src/develop/masks/circle.c

```diff
@@ -942,21 +941,25 @@ static void _circle_events_post_expose(cairo_t *cr,
       else
         cangle = -(M_PI / 2) - cangle;
 
-      // (arrowx,arrowy) is the point of intersection, we move it
-      // (factor 1.11) a bit farther than the inner circle to avoid
-      // superposition.
-      const float arrowx = gpt->points[0] + 1.11 * radius * cosf(cangle);
-      const float arrowy = gpt->points[1] + 1.11 * radius * sinf(cangle);
-
-      cairo_move_to(cr, gpt->source[0], gpt->source[1]); // source center
-      cairo_line_to(cr, arrowx, arrowy);                 // dest border
-      // then draw to line for the arrow itself
-      const float arrow_scale = 6.0f * pr_d;
-      cairo_move_to(cr, arrowx + arrow_scale * cosf(cangle + (0.4f)),
-                    arrowy + arrow_scale * sinf(cangle + (0.4f)));
-      cairo_line_to(cr, arrowx, arrowy);
-      cairo_line_to(cr, arrowx + arrow_scale * cosf(cangle - (0.4f)),
-                    arrowy + arrow_scale * sinf(cangle - (0.4f)));
+      // (to_x, to_y) is the point of intersection
+      const float to_x = gpt->points[0] + radius * cosf(cangle);
+      const float to_y = gpt->points[1] + radius * sinf(cangle);
+
```

Comment:
note : 
- (a very old omission I just spot) that only works without any distortion, we may want to apply distortion to the resulting point... I've not checked, but chance are that this also apply to the other shapes !
- same computation can also be done with source point. That may gain some cpu cyles...

Reply:
> * (a very old omission I just spot) that only works without any distortion, we may want to apply distortion to the resulting point... I've not checked, but chance are that this also apply to the other shapes !

Fixed. And no this is not the case for other shapes which are using the distorted set of points.